### PR TITLE
Added resetTimer function to Timer

### PR DESCRIPTION
**Pull request by SofieVonge**
Added resetTimer function to Timer plus three tests; two in getElapsedTime and one in resetTimer

### CODE REVIEW

File: lib/Timer/Timer.test.js [hunk 1]
```diff
@@ -89,6 +89,25 @@ describe('Timer', () => {
 
       expect(timer.getElapsedTime()).toBe(timeElapsed)
     })
+
+    it('Should return 0 when resetTimer has been called on a new timer', () => {
+      const timer = new Timer()
+
+      timer.startTimer()
```
Comment (goibon) [inline lib/Timer/Timer.test.js]: This call seems redundant if you want to test a new "fresh" timer 🤔

File: lib/Timer/Timer.test.js [hunk 2]
```diff
@@ -165,4 +184,15 @@ describe('Timer', () => {
       })
     })
   })
+  describe('resetTimer', () => {
+    it('Should clear associated interval', () => {
```
Comment (goibon) [inline lib/Timer/Timer.test.js:187]: Should we also have a test that assert that `startTime` and `endTime` get cleared?

File: lib/Timer/index.js [hunk 3]
```diff
@@ -29,6 +29,12 @@ class Timer {
     this.intervalId = undefined
   }
 
+  resetTimer () {
+    clearInterval(this.intervalId)
+    this.intervalId = undefined
+    this.startTime = undefined
```
Comment (goibon) [inline lib/Timer/index.js:35]: Shouldn't it also clear `this.endTime`?

**Review by goibon — COMMENTED** [review]
While you're at it, perhaps you should add tests to `DecrementingTimer` which extends the `Timer` class and make sure that its `getRemainingTime` function returns the expected value when calling reset 💁

File: lib/Timer/index.js (hunk 3 above)
Comment (SofieVonge) [reply lib/Timer/index.js:35]: Haha yes, I don't know why I thought that you would never try to end the timer before resetting it.

**Pull request merged**